### PR TITLE
Simplification of the dbt Integration tutorial

### LINKED ISSUE #2356: Integration simplification: dbt core / Cosmos
**Request type:**

- [x] **Addition to an existing guide**: Add the `area-learn` and `enhancement` labels to this issue.

**Use case or feature that should be covered:**

Short tutorial showing how to use Airflow with dbt core using Cosmos. As discussed in the meeting today, task group front and center, no SDK, short example.

**Who the guide is intended for:**
Users who want to know how to use Airflow with dbt core.

**Are you the right person to review a draft of the guide for technical accuracy? If not, who is?**
@jlaneve

### PR DESCRIPTION
**Pull request by TJaniF**
Resolves #2356 

As discussed earlier this week focussing the integration tutorial on the Astro dbt provider. The full use case tutorial will eventually live in Examples :)

### CODE REVIEW

File: learn/airflow-dbt.md [hunk 1]
```diff
@@ -39,153 +60,83 @@ To get the most out of this tutorial, make sure you have an understanding of:
...
+- Access to a data warehouse supported by dbt Core. See [dbt documentation](https://docs.getdbt.com/docs/supported-data-platforms) for all supported warehouses. This tutorial uses a Postgres database.
 
 You do not need to have dbt Core installed locally in order to complete this tutorial.
 
 ## Step 1: Configure your Astro project
 
-An Astro project contains all of the files you need to run Airflow locally.
+To use dbt with Airflow install dbt Core in a virtual environment and the Astro dbt provider in your Astro project.
 
-1. Create a new Astro project:
```
Comment (kentdanas) [inline learn/airflow-dbt.md:54]: We should leave this in. Generally we want to simplify the example DAGs shown in these tutorials to (wherever possible) only use the tool in question. But, we still need to provide context around setting up the project, etc.

File: learn/airflow-dbt.md [hunk 2]
```diff
@@ -39,153 +60,83 @@ To get the most out of this tutorial, make sure you have an understanding of:
...
-CREATE SCHEMA energy_schema;
-```
-
-If you are using a different data warehouse, your commands to create the database and schema may differ.
+## Step 2: Prepare your dbt project
 
-## Step 4: Create your dbt models
-
-In this tutorial we will use an example dbt job that consists of two dbt models. The first model, called `select_country`, will get the data for a selected country. The second model, called `create_pct`, will use the table created by the first model to calculate the percentage of renewable and solar energy capacity in that country.
+In order to integrate your dbt project with Airflow, you need to add the dbt project in a folder called `dbt` in the `dags` folder. You can either add your own project or follow the steps below to create a simple project using two models.
```
Comment (kentdanas) [inline learn/airflow-dbt.md:94]:
```suggestion
In order to integrate your dbt project with Airflow, you need to add your dbt project in a subfolder called `dbt` within your `dags` folder. You can either add your own project, or follow the steps below to create a simple project using two models.
```

File: learn/airflow-dbt.md [hunk 3]
```diff
@@ -39,153 +60,83 @@ To get the most out of this tutorial, make sure you have an understanding of:
...
+    └── dbt
+        └── my_simple_dbt_project
+           ├── dbt_project.yml
+           └── models
+               ├── model1.sql
+               └── model2.sql
 ```
 
-## Step 5: Configure a data warehouse connection
+## Step 3: Create an Airflow connection to your data warehouse
```
Comment (kentdanas) [inline learn/airflow-dbt.md:137]: I think totally fine to leave this in here even if we have a connections guide. As I mentioned above, we still want to provide the setup and context throughout the tutorial, and want to limit how much people need to click into other docs to find the info they need.

File: learn/airflow-dbt.md [hunk 4]
```diff
@@ -207,43 +158,29 @@ If a connection type for your database isn't available, you might need to make i
 
 :::
 
-## Step 6: Write your Airflow DAG
+## Step 4: Write your Airflow DAG
 
-The DAG you'll write uses the Astro dbt provider to create tasks from existing dbt models and embeds those tasks within other actions in your data ecosystem.
+The DAG you'll write uses the Astro dbt provider to create tasks from existing dbt models. You can add up- and downstream tasks to and embeds the dbt project within other actions in your data ecosystem.
```
Comment (kentdanas) [inline learn/airflow-dbt.md]:
```suggestion
The DAG you'll write uses the Astro dbt provider to create tasks from existing dbt models. You can also add up- and downstream tasks to embed the dbt project within other actions in your data ecosystem.
```
Double check me on this edit - I _think_ this is what you were trying to say?

File: learn/airflow-dbt.md [hunk 5]
```diff
@@ -207,43 +158,29 @@ If a connection type for your database isn't available, you might need to make i
...
-
-
 :::info
 
 The DbtTaskGroup class populates an Airflow task group with Airflow tasks created from dbt models inside of a normal DAG. To directly define a full DAG containing only dbt models use the `DbtDag` class, as shown in the [Astro dbt provider documentation](https://astronomer.github.io/astronomer-cosmos/dbt/usage.html#full-dag).
 
 :::
 
-
 ## Alternative ways to run dbt Core with Airflow
```
Comment (kentdanas) [inline learn/airflow-dbt.md:184]: Do we think it would make sense to put this section below the Conclusion? Obviously "Conclusion" seems to make the most sense at the end, but it seems a little weird to talk about alternatives and then go back to say congrats for using Cosmos... or maybe we we title "conclusion" something else...

**Review by kentdanas — APPROVED** [review]
Looks good to me! A few minor comments, but I'm preemptively approving - good to move on to copy edit.

Also can update naming and close out #2331 with this.

File: learn/airflow-dbt.md (hunk 5 above)
Comment (TJaniF) [reply learn/airflow-dbt.md:203]: Hmmm... yeah, I think back when we start adding the conclusion we barely had tutorials with appendix information like this. I vote for putting it before the alternative ways.... it feels more natural to me than expanding it to say "we also congratulate you for learning xyz" 😄 

How about calling it "Next steps" ? (I feel a slight deja vu, sorry if we discussed that before)

**Review by TJaniF — COMMENTED** [review]
Thank you @kentdanas :) 

Good callout on how the code snippet is not exactly a quickstart! I like your suggested version much better.

@lzdanski and @jwitz ready for you :)

File: learn/airflow-dbt.md [hunk 6]
```diff
@@ -1,35 +1,64 @@
...
+        "schema": SCHEMA_NAME,
+        "vars": '{"my_key": "my_value"}',
+    },
+)
+```
 
-For a tutorial on how to use dbt Cloud with Airflow see [Orchestrate dbt Cloud with Airflow](airflow-dbt-cloud.md).
+:::info
+
+If you are already familiar with Airflow and dbt Core check out this [github repository](https://github.com/astronomer/astro-dbt-provider-tutorial-example).
```
Comment (jwitz) [inline learn/airflow-dbt.md:37]: This doesn't explain what the repo is or why I would want to check it out. Maybe reword to be "If you are familiar with Airflow and dbt Core and just want to get your project running, copy the example project from [Astronomer GitHub](https://github.com/astronomer/astro-dbt-provider-tutorial-example) and run it locally using the Astro CLI."

File: learn/airflow-dbt.md [hunk 7]
```diff
@@ -1,35 +1,64 @@
...
+
+:::
 
 :::info
 
-The full Astro project used in this tutorial can be cloned from [this repository](https://github.com/astronomer/astro-dbt-provider-tutorial-example). 
+For a tutorial on how to use dbt Cloud with Airflow see [Orchestrate dbt Cloud with Airflow](airflow-dbt-cloud.md).
```
Comment (jwitz) [inline learn/airflow-dbt.md:43]:
```suggestion
For a tutorial on how to use dbt Cloud with Airflow, see [Orchestrate dbt Cloud with Airflow](airflow-dbt-cloud.md).
```

File: learn/airflow-dbt.md [hunk 8]
```diff
@@ -1,35 +1,64 @@
...
 
 :::
 
+## Why use Airflow with dbt Core?
+
+dbt Core offers the possibility to build modular, reuseable SQL components with built-in dependency management and [incremental builds](https://docs.getdbt.com/docs/build/incremental-models). With [Cosmos](https://astronomer.github.io/astronomer-cosmos/) you can integrate dbt jobs into your Airflow orchestration environment as a standalone DAG or as a task group within a DAG. 
+
+The benefits of using Airflow with dbt Core include:
+
+- Use Airflow's [data-aware scheduling](airflow-datasets.md) and [Airflow sensors](what-is-a-sensor.md) to run models depending on any other event in your data ecosystem.
```
Comment (jwitz) [inline learn/airflow-dbt.md:53]:
```suggestion
- Use Airflow's [data-aware scheduling](airflow-datasets.md) and [Airflow sensors](what-is-a-sensor.md) to run models depending on other events in your data ecosystem.
```

File: learn/airflow-dbt.md [hunk 9]
```diff
@@ -1,35 +1,64 @@
...
+- Turn each dbt model into a task, complete with Airflow features like [retries](rerunning-dags.md#automatically-retry-tasks) and [error notifications](error-notifications-in-airflow.md), as well as full observability into past runs directly in the Airflow UI.
+- Run `dbt test` on tables created by individual models immediately after a model has completed. Catch issues before moving downstream and integrate additional [data quality checks](data-quality.md) with your preferred tool to run alongside dbt tests.
+- Run dbt projects using [Airflow connections](connections.md) instead of dbt profiles. You can store all your connections in one place, directly within Airflow or by using a [secrets backend](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/security/secrets/secrets-backend/index.html).
+- Leverage native support for installing and running dbt in a virtual environment to avoid dependency conflicts with Airflow.
+
 ## Time to complete
 
-This tutorial takes approximately 1 hour to complete.
+This tutorial takes approximately 30 minutes to complete.
```
Comment (jwitz) [inline learn/airflow-dbt.md:61]: Nice 😎

File: learn/airflow-dbt.md [hunk 10]
```diff
@@ -58,134 +88,70 @@ An Astro project contains all of the files you need to run Airflow locally.
...
-CREATE SCHEMA energy_schema;
-```
-
-If you are using a different data warehouse, your commands to create the database and schema may differ.
+## Step 2: Prepare your dbt project
 
-## Step 4: Create your dbt models
-
-In this tutorial we will use an example dbt job that consists of two dbt models. The first model, called `select_country`, will get the data for a selected country. The second model, called `create_pct`, will use the table created by the first model to calculate the percentage of renewable and solar energy capacity in that country.
+In order to integrate your dbt project with Airflow, you need to add your dbt project in a subfolder called `dbt` within your `dags` folder. You can either add your own project, or follow the steps below to create a simple project using two models.
```
Comment (jwitz) [inline learn/airflow-dbt.md:109]:
```suggestion
To integrate your dbt project with Airflow, you need to add your dbt project to a subfolder called `dbt` within your `dags` folder. You can either add your own project, or follow the steps below to create a simple project using two models.
```

File: learn/airflow-dbt.md [hunk 11]
```diff
@@ -58,134 +88,70 @@ An Astro project contains all of the files you need to run Airflow locally.
...
-    # create a variable called country_code and give it the default value "FR" (for France)
+    ```yaml
+    name: 'my_simple_dbt_project'
     vars:
-      country_code: "FR"
+        my_name: "No entry"
     ```
 
-4. In `my_energy_project`, create a folder called `models`.
+4. Add your dbt models in a folder called `models` within the `my_simple_dbt_project` folder. You can add as many models as you want to run. This tutorial uses the following two models:
```
Comment (jwitz) [inline learn/airflow-dbt.md:123]:
```suggestion
4. Add your dbt models in a subfolder called `models` in the `my_simple_dbt_project` folder. You can add as many models as you want to run. This tutorial uses the following two models:
```

File: learn/airflow-dbt.md [hunk 12]
```diff
@@ -58,134 +88,70 @@ An Astro project contains all of the files you need to run Airflow locally.
...
+           ├── dbt_project.yml
+           └── models
+               ├── model1.sql
+               └── model2.sql
 ```
 
-## Step 5: Configure a data warehouse connection
+## Step 3: Create an Airflow connection to your data warehouse
+
+Cosmos allows you to use Airflow connections inside your dbt project.
```
Comment (jwitz) [inline learn/airflow-dbt.md:154]:
```suggestion
Cosmos allows you to apply Airflow connections to your dbt project. 
```
How would you connect to the database from a non-integrated dbt project? Does Airflow make that process easier? We could mention that if so.

File: learn/airflow-dbt.md [hunk 13]
```diff
@@ -207,46 +173,36 @@ If a connection type for your database isn't available, you might need to make i
 
 :::
 
-## Step 6: Write your Airflow DAG
+## Step 4: Write your Airflow DAG
 
-The DAG you'll write uses the Astro dbt provider to create tasks from existing dbt models and embeds those tasks within other actions in your data ecosystem.
+The DAG you'll write uses Cosmos to create tasks from existing dbt models. You can add up- and downstream tasks to embed the dbt project within other actions in your data ecosystem.
```
Comment (jwitz) [inline learn/airflow-dbt.md:178]:
```suggestion
The DAG you'll write uses Cosmos to create tasks from existing dbt models. You can add upstream and downstream tasks to embed the dbt project within other actions in your data ecosystem.
```

File: learn/airflow-dbt.md [hunk 14]
```diff
@@ -207,46 +173,36 @@ If a connection type for your database isn't available, you might need to make i
...
-
-
 :::info
 
-The DbtTaskGroup class populates an Airflow task group with Airflow tasks created from dbt models inside of a normal DAG. To directly define a full DAG containing only dbt models use the `DbtDag` class, as shown in the [Astro dbt provider documentation](https://astronomer.github.io/astronomer-cosmos/dbt/usage.html#full-dag).
+The DbtTaskGroup class populates an Airflow task group with Airflow tasks created from dbt models inside of a normal DAG. To directly define a full DAG containing only dbt models use the `DbtDag` class, as shown in the [Cosmos documentation](https://astronomer.github.io/astronomer-cosmos/dbt/usage.html#full-dag).
 
 :::
 
+## Next steps
```
Comment (jwitz) [inline learn/airflow-dbt.md:199]:
```suggestion
```

Comment (jwitz) [reply learn/airflow-dbt.md:199]: I don't think we need a header if there's not an actual "Next step" to take here

File: learn/airflow-dbt.md [hunk 15]
```diff
@@ -1,35 +1,64 @@
...
+```
 
-For a tutorial on how to use dbt Cloud with Airflow see [Orchestrate dbt Cloud with Airflow](airflow-dbt-cloud.md).
+:::info
+
+If you are already familiar with Airflow and dbt Core check out this [github repository](https://github.com/astronomer/astro-dbt-provider-tutorial-example).
+
+:::
 
 :::info
```
Comment (jwitz) [inline learn/airflow-dbt.md:42]:
```suggestion
```

File: learn/airflow-dbt.md [hunk 16]
```diff
@@ -1,35 +1,64 @@
...
 
-The full Astro project used in this tutorial can be cloned from [this repository](https://github.com/astronomer/astro-dbt-provider-tutorial-example). 
+For a tutorial on how to use dbt Cloud with Airflow see [Orchestrate dbt Cloud with Airflow](airflow-dbt-cloud.md).
 
 :::
```
Comment (jwitz) [inline learn/airflow-dbt.md:46]:
```suggestion
```

**Review by jwitz — APPROVED** [review]
Looks great @TJaniF ! Just a few minor suggestions

File: learn/airflow-dbt.md [hunk 17]
```diff
@@ -1,35 +1,64 @@
 ---
-title: "Orchestrate dbt Core jobs with the Astro dbt provider"
+title: "Orchestrate dbt Core jobs with Cosmos"
```
Comment (jwitz) [inline learn/airflow-dbt.md:2]:
```suggestion
title: "Orchestrate dbt Core jobs with Airflow and Cosmos"
```
Just in case people come from google search, we should still make it clear that this is an Airflow-first guide

File: learn/airflow-dbt.md (hunk 12 above)
Comment (TJaniF) [reply learn/airflow-dbt.md]: If you would be running dbt by itself, not with Airflow? You would define the connection in a profiles.yaml file: https://docs.getdbt.com/docs/core/connect-data-platform/connection-profiles 
The info that has to be provided is the same, so not necessarily easier, the upside of using Cosmos is to be able to hand the information from the Airflow connection to the dbt project, so you only need to define and maintain the connection in one place.

File: learn/airflow-dbt.md (hunk 6 above)
Comment (TJaniF) [reply learn/airflow-dbt.md]: Yeah, good point. does this work:

> If you are already familiar with Airflow and dbt Core and just want to get a project running, clone [Astronomers Cosmos example repository](https://github.com/astronomer/astro-dbt-provider-tutorial-example) and run it locally using the Astro CLI. 


? :)

**Review by TJaniF — COMMENTED** [review]
Thank you for the review @jwitz ! 

Accepted all the suggestions and rephrased the sentence linking to the old tutorial repository :)

**Review by jlaneve — APPROVED** [review]
